shared/utils/relabel-ids.py

```diff
@@ -304,32 +303,35 @@ def cce_is_valid(cceid):
     return match is not None
 
 
+def definition_extends_nonexisting_checks(definition, indexed_oval_defs):
+    for extdefinition in definition.findall(".//{%s}extend_definition" % oval_ns):
+        # Verify each extend_definition in the definition
+        extdefinitionref = extdefinition.get("definition_ref")
+
+        # Search the OVAL tree for a definition with the referred ID
+        referreddefinition = indexed_oval_defs.get(extdefinitionref)
+
+        if referreddefinition is None:
+            # There is no oval satisfying the extend_definition referal
+            print("WARNING: OVAL definition '{0}' extends non-existing '{1}', "
+                  "removing it from OVAL definitions."
+                  .format(definition.get("id"), extdefinitionref),
+                  file=sys.stderr)
+            return True
+    return False
+
+
 def drop_oval_checks_extending_non_existing_checks(ovaltree, indexed_oval_defs):
     # Incomplete OVAL checks are as useful as non existing checks
     # Here we check if all extend_definition refs from a definition exists in local OVAL file
     #
     # TODO: handle multiple levels of referrals.
     # OVAL checks that go beyond one level of extend_definition won't be completely removed
     definitions = ovaltree.find(".//{%s}definitions" % oval_ns)
-    defstoremove = []
+    defstoremove = set()
     for definition in definitions:
-        incomplete = False;
-        for extdefinition in definition.findall(".//{%s}extend_definition" % oval_ns):
-            # Verify each extend_definition in the definition
-            extdefinitionref = extdefinition.get("definition_ref")
-
-            # Search the OVAL tree for a definition with the referred ID
-            referreddefinition = indexed_oval_defs.get(extdefinitionref)
-
-            if referreddefinition is None:
-                # There is no oval satisfying the extend_definition referal
-                incomplete = True
-                print("WARNING: OVAL definition '{0}' extends non-existing '{1}', "
-                      "removing it from OVAL definitions."
-                      .format(definition.get("id"), extdefinitionref),
-                      file=sys.stderr)
-        if incomplete:
-            defstoremove.append(definition)
+        if definition_extends_nonexisting_checks(definition, indexed_oval_defs):
+            defstoremove.add(definition)
```

Comment:
I would expect the warning to be printed here and not in the function. If I am calling a function asking about some state I don't want it to write out messages.